src/test/java/com/unb/tracker/TrackerApplicationTests.java

```diff
@@ -1,107 +1,183 @@
 
 import java.sql.Date;
 import java.text.DateFormat;
 import java.text.SimpleDateFormat;
+import java.util.ArrayList;
 import java.util.Arrays;
+import java.util.List;
+import java.util.regex.Pattern;
 
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.AdditionalAnswers.returnsFirstArg;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static org.springframework.security.test.web.servlet.request.SecurityMockMvcRequestBuilders.formLogin;
+import static org.springframework.security.test.web.servlet.response.SecurityMockMvcResultMatchers.authenticated;
+import static org.springframework.security.test.web.servlet.setup.SecurityMockMvcConfigurers.springSecurity;
 import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
 import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
 import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
-import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
-import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.view;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
 
 
 @RunWith(SpringRunner.class)
 @SpringBootTest
-@AutoConfigureMockMvc
 public class TrackerApplicationTests {
     protected final DateFormat dateFormat = new SimpleDateFormat("dd-MM-yyyy");
 
     @Autowired
-	private TrackerController controller;
+    private CourseController courseController;
+
     @Autowired
-    private MockMvc mockMvc;
+    private UserController userController;
+
     @Autowired
+    private WebApplicationContext context;
+
+    private MockMvc mockMvc;
+
+    @MockBean
     private CourseRepository courseRepository;
 
+    @MockBean
+    private UserRepository userRepository;
+
+    //TODO:
+            //- Loading the index(login) should render
+            //- Selecting one of the buttons should render the login card
+            //- Selecting sign up should render the sign up card
+            //- logging in with invalid credentials should redirect to index with error message
+            //- attempting to create an account without an email address, username or password should redirect to index with error message
+            //- test username and password validation (unique username with 6 to 32 length, matching passwords with length 8 to 32)
+            //- Creating a new user should correctly store their information in the database
+            //- Creating a new student should redirect to their student view
+            //- Creating a new instructor should redirect to their instructor view
+            //- Attempting to login as instructor or user should redirect them correctly
+
+    @Before
+    public void setup() {
```

Comment:
I asssume this is the replacement of the `@AutoConfigureMockMvc`

Reply:
Yup! Spring needs a bit of help to set up security tests!